build.rs

```diff
@@ -33,13 +36,25 @@ fn download_and_unzip(client: &reqwest::blocking::Client, url: &str) {
 
 #[cfg(feature = "build")]
 fn main() {
+    let target = env::var("TARGET").unwrap();
+    let mut config = cpp_build::Config::new();
+
     println!("cargo:rerun-if-changed=./files");
 
+    //only for windows mingw64/gnu tool chain
+    if target.contains("windows-gnu") {
+        config.flag("-Os");
+        config.flag("-Wa,-mbig-obj");
+
+        let root_dir = std::env::var("CARGO_MANIFEST_DIR").unwrap();
+        let libs = PathBuf::from(root_dir).join("external-libs").join("windows");
+
+        println!("cargo:rustc-flags=-L '{}'", libs.display());
+        println!("cargo:rustc-link-lib={}", "blas");
+        println!("cargo:rustc-link-lib={}", "lapack");
+    }
     println!("cargo:rustc-link-lib=dlib");
-    println!("cargo:rustc-link-lib=blas");
```

Comment:
blas, lapack: The other OSes (such as `Linux`) will be broken.

Reply:
I ll try to bundle the blas, lapack for amd64 precompliled binaries instead of depending on OS libs.